### ISSUE TITLE
Build Issues on OSX (Need to test linux)

### ISSUE THREAD

**Issue by jlaura** [USGS-Astrogeology/swigcsm#14]
This needs to also tested on linux to see if this is reproducible there as well.  I have conda installed the `csm` package from conda-forge and am using the tentative conda-forge build for the 1.1.x dev release slated for the end of this week. Both libraries build without issue on OSX (either in CI for the former or locally for the latter). All @rpath's are working as expected and ctest testing is passing.

When trying to do a conda build of the swigcsm library, I get a clean build but errors either in linking or straight up segmentation faults.  

**Linking Issues**
I think that the linking issues are conda build related when we have a mismatch between the python version in the root conda env and the python version being used in the build env. This is manifesting as linking errors during the build. I think this because:

<details>
  <summary>conda build output:</summary> 

```bash
$ conda build --python 3.5 pycsm/
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
WARNING:conda_build.metadata:No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
Adding in variants from internal_defaults
INFO:conda_build.variants:Adding in variants from internal_defaults
Adding in variants from config.variant
INFO:conda_build.variants:Adding in variants from config.variant
Attempting to finalize metadata for pycsm
INFO:conda_build.metadata:Attempting to finalize metadata for pycsm
Solving environment: ...working... done
Solving environment: ...working... done
Solving environment: ...working... done
BUILD START: ['pycsm-1.0.0-py35_0.tar.bz2']
Solving environment: ...working... done

## Package Plan ##

  environment location: /Users/jlaura/miniconda3/conda-bld/pycsm_1550524113289/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla


The following NEW packages will be INSTALLED:

    bzip2:           1.0.6-h1de35cc_1002   conda-forge
    ca-certificates: 2018.11.29-ha4d7672_0 conda-forge
    certifi:         2018.8.24-py35_1001   conda-forge
    csm:             v3.0.3.1-1001         local
    libcxx:          7.0.0-h2d50403_2      conda-forge
    libffi:          3.2.1-h0a44026_1005   conda-forge
    llvm-meta:       7.0.0-0               conda-forge
    ncurses:         6.1-h0a44026_1002     conda-forge
    openssl:         1.0.2p-h1de35cc_1002  conda-forge
    pip:             18.0-py35_1001        conda-forge
    python:          3.5.5-h5001a0f_2      conda-forge
    readline:        7.0-hcfe32e1_1001     conda-forge
    setuptools:      40.4.3-py35_0         conda-forge
    sqlite:          3.26.0-h1765d9f_1000  conda-forge
    tk:              8.6.9-ha441bb4_1000   conda-forge
    wheel:           0.32.0-py35_1000      conda-forge
    xz:              5.2.4-h1de35cc_1001   conda-forge
    zlib:            1.2.11-h1de35cc_1004  conda-forge

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Solving environment: ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /Users/jlaura/miniconda3/conda-bld/pycsm_1550524113289/_build_env


The following NEW packages will be INSTALLED:

    bzip2:           1.0.6-h1de35cc_1002   conda-forge
    ca-certificates: 2018.11.29-ha4d7672_0 conda-forge
    certifi:         2018.11.29-py37_1000  conda-forge
    csm:             v3.0.3.1-1001         local
    libcxx:          7.0.0-h2d50403_2      conda-forge
    libffi:          3.2.1-h0a44026_1005   conda-forge
    llvm-meta:       7.0.0-0               conda-forge
    ncurses:         6.1-h0a44026_1002     conda-forge
    openssl:         1.0.2p-h1de35cc_1002  conda-forge
    pcre:            8.41-h0a44026_1003    conda-forge
    pip:             19.0.2-py37_0         conda-forge
    python:          3.7.1-h145921a_1000   conda-forge
    readline:        7.0-hcfe32e1_1001     conda-forge
    setuptools:      40.8.0-py37_0         conda-forge
    sqlite:          3.26.0-h1765d9f_1000  conda-forge
    swig:            3.0.12-h0a44026_1003  conda-forge
    tk:              8.6.9-ha441bb4_1000   conda-forge
    wheel:           0.33.0-py37_0         conda-forge
    xz:              5.2.4-h1de35cc_1001   conda-forge
    zlib:            1.2.11-h1de35cc_1004  conda-forge
```
</details>

Note how the `_h_env_placeholder...` and `_build_env` environments have different version of python. I think that this is causing this error:

```
ImportError: dlopen(/<path to me>/conda-bld/pycsm_1550524113289/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/python3.5/site-packages/csmapi/_csmapi.so, 2): Library not loaded: @rpath/libpython3.7m.dylib
  Referenced from: /<path to me>/conda-bld/pycsm_1550524113289/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/python3.5/site-packages/csmapi/_csmapi.so
  Reason: image not found
```

I also see: `ERROR: /<path to me>/conda-bld/pycsm_1550524113289/_build_env/lib/libpython3.7m.dylib found in build prefix; should never happen` which needs to be tracked down.

**Segmentation Faults** 

Given the above, I updated to Python3.7 in my root env so that the build env would match. This got a build to work (no rpath issues on the test phase of conda build), but resulted in:

```export PREFIX=/<path to me>/conda-bld/pycsm_1550534514415/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_
export SRC_DIR=/<path to me>/conda-bld/pycsm_1550534514415/test_tmp
/Users/jlaura/miniconda3/conda-bld/pycsm_1550534514415/test_tmp/conda_test_runner.sh: line 2: 50852 Segmentation fault: 11  "/<path to me>/conda-bld/pycsm_1550534514415/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/bin/python" -s "/Users/jlaura/miniconda3/conda-bld/pycsm_1550534514415/test_tmp/run_test.py"
Tests failed for pycsm-1.0.0-py37_0.tar.bz2 - moving package to /<path to me>/conda-bld/broken
```

Looking inside the tarball, the build looks clean, rpaths are all good, so I am not sure where this segmentation fault is coming from.

**Comment by jlaura** [comment]
@jessemapel @Kelvinrr I looped you all in to see if you can also reproduce on OSX. Low priority given our current work, but still something that is a blocker for our CTX work in a few weeks.

**Comment by jlaura** [comment]
I can confirm that, building the usgscsm and pycsm packages manually (outside of conda) on linux results in a fully functional wrapper. Next test is to build both inside of conda.

**Comment by jlaura** [comment]
```
(lldb) settings set -- target.run-args  "test.py"
(lldb) run
Process 5280 launched: '/Users/jlaura/miniconda3/envs/rollback/bin/python' (x86_64)
Process 5280 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x10)
    frame #0: 0x00000001013040f7 libpython3.7m.dylib`PyModule_Create2 + 7
libpython3.7m.dylib`PyModule_Create2:
->  0x1013040f7 <+7>:  movq   0x10(%rax), %rax
    0x1013040fb <+11>: cmpq   $0x0, 0x28(%rax)
    0x101304100 <+16>: je     0x101304107               ; <+23>
    0x101304102 <+18>: jmp    0x101304120               ; _PyModule_CreateInitialized
Target 0: (python) stopped.
```

**Comment by jlaura** [comment]
closed by #13
